int16 current_position
int16 next_to_location